miasm2/ir/translators/C.py

```diff
@@ -194,38 +284,123 @@ def from_ExprOp(self, expr):
                     self._size2mask(expr.args[0].size)
                 )
             elif expr.op in self.dct_rot:
-                return '(%s(%s, %s, %s) &%s)' % (
-                    self.dct_rot[expr.op],
-                    expr.args[0].size,
-                    self.from_expr(expr.args[0]),
-                    self.from_expr(expr.args[1]),
-                    self._size2mask(expr.args[0].size),
-                )
+                arg0 = self.from_expr(expr.args[0])
+                arg1 = self.from_expr(expr.args[1])
+                if expr.size <= self.NATIVE_INT_MAX_SIZE:
+                    out = '(%s(%s, %s, %s) &%s)' % (
+                        self.dct_rot[expr.op],
+                        expr.args[0].size,
+                        self.from_expr(expr.args[0]),
+                        self.from_expr(expr.args[1]),
+                        self._size2mask(expr.args[0].size),
+                    )
+                else:
+                    op = {
+                        ">>>": "ror",
+                        "<<<": "rol"
+                    }
+                    out = "bignum_%d(%s, %d, bignum_to_uint64(%s))" % (
+                        op[expr.op], arg0, expr.size, arg1
+                    )
+                    out = "bignum_mask(%s, %d)"% (out, expr.size)
+                return out
+
             elif expr.op == 'x86_cpuid':
                 return "%s(%s, %s)" % (expr.op,
                                        self.from_expr(expr.args[0]),
                                        self.from_expr(expr.args[1]))
             elif (expr.op.startswith("fcom")  or
                   expr.op in ["fadd", "fsub", "fdiv", 'fmul', "fscale",
                               "fprem", "fprem_lsb", "fyl2x", "fpatan"]):
-                return "fpu_%s%d(%s, %s)" % (
-                    expr.op,
-                    expr.size,
-                    self.from_expr(expr.args[0]),
-                    self.from_expr(expr.args[1]),
-                )
+                arg0 = self.from_expr(expr.args[0])
+                arg1 = self.from_expr(expr.args[1])
+                if not expr.args[0].size <= self.NATIVE_INT_MAX_SIZE:
```

Comment:
If the argument is > 64 bit, the operation must fail. If I guess correctly that this check is for case when `NATIVE_INT_MAX_SIZE` is less than 64, an additional assert would avoid a silent fail

Reply:
Done